### ISSUE TITLE
`cmp_owned` triggers when the comparison requires conversion

### ISSUE THREAD

**Issue by Jarcho** [rust-lang/rust-clippy#12494]
Types like `proc_macro::Ident` can only have their value viewed by converting them to a string first.

### Reproducer

I tried this code:

```rust
fn f(i: proc_macro2::Ident) -> bool {
  i.to_string() == "foo"
}
```

I saw this happen:

```
warning: this creates an owned instance just for comparison
 --> src/main.rs:2:3
  |
2 |   i.to_string() == "foo"
  |   ^^^^^^^^^^^^^ help: try: `i`
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#cmp_owned
  = note: `#[warn(clippy::cmp_owned)]` on by default
```

### Version

```text
Playground 2024-03-14
```

**Comment by Jarcho** [comment]
Looks like `proc_macro2` actually does let you compare directly.
